sdk/core/azure-core/src/main/java/com/azure/core/implementation/ImplUtils.java

```diff
@@ -318,6 +320,21 @@ public static String bomAwareToString(byte[] bytes, int offset, int count, Strin
         }
     }
 
+    /**
+     * Creates a new {@link URL} from the given {@code urlString}.
+     * <p>
+     * This is a temporary method that will be removed once all usages of {@link URL#URL(String)} are migrated to
+     * {@link URI}-based methods given the deprecation of the URL methods in Java 20.
+     *
+     * @param urlString The string to convert to a {@link URL}.
+     * @return The {@link URL} representing the {@code urlString}.
+     * @throws MalformedURLException If the {@code urlString} isn't a valid {@link URL}.
+     */
+    @SuppressWarnings("deprecation")
+    public static URL createUrl(String urlString) throws MalformedURLException {
```

Comment:
Why do we need this method? Is this intended to be the util method for all string->URL conversions when we switch to URI?

Reply:
Yeah, I thought it'd be good to centralize all the one-off `URL` creations into a single API we can manage better